common/reconciliation/invariants/invariantManager.go

```diff
@@ -127,14 +128,15 @@ func (i *invariantManager) nextCheckResultType(
 func flattenInvariants(
 	collections []common.InvariantCollection,
 	pr common.PersistenceRetryer,
+	scanType common.ScanType,
 ) ([]common.Invariant, []common.InvariantType) {
 	var ivs []common.Invariant
 	for _, collection := range collections {
 		switch collection {
 		case common.InvariantCollectionHistory:
-			ivs = append(ivs, getHistoryCollection(pr)...)
+			ivs = append(ivs, getHistoryCollection(pr, scanType)...)
 		case common.InvariantCollectionMutableState:
-			ivs = append(ivs, getMutableStateCollection(pr)...)
+			ivs = append(ivs, getMutableStateCollection(pr, scanType)...)
 		}
```

Comment:
How about creating a new invariant collection specifically for current mutable state records? I think it'll be easier for adding new checks for current records and `scanType` parameter won't be necessary.

Reply:
This class will be gone with the refactoring phase, as discussed, and so the parameter here as well.